server/src/watchdog.js

```diff
@@ -0,0 +1,141 @@
+const config = require("./config").getProperties();
+const mozlog = require("./logging").mozlog("server");
+const db = require("./db");
+const { Shot } = require("./servershot");
+const util = require("util");
+const { URL } = require("url");
+const fetch = require("node-fetch");
+const Hawk = require("hawk");
+const FormData = require("form-data");
+const genUuid = require("nodify-uuid");
+
+const isEnabled = config.watchdog.enable;
+const isProperlyConfigured = validateConfiguration();
+
+function validateConfiguration() {
+  if (!isEnabled) {
+    // Do no warn about config values if Watchdog is not enabled.
+    return false;
+  }
+
+  let isConfigValid = true;
+
+  if (!config.watchdog.id || !config.watchdog.key || !config.watchdog.algorithm) {
+    mozlog.warn("watchdog-invalid-config", {msg: "Watchdog credentials configuration values are missing."});
+    isConfigValid = false;
+  }
+
+  if (!config.watchdog.submissionUrl) {
+    mozlog.warn("watchdog-invalid-config", {msg: "Watchdog submission URL configuration value is missing."});
+    isConfigValid = false;
+  } else {
+    try {
+      new URL(config.watchdog.submissionUrl);
+    } catch (e) {
+      mozlog.warn("watchdog-invalid-config", {msg: "Watchdog submission URL is not a valid URL.", err: e});
+      isConfigValid = false;
+    }
+  }
+
+  return isConfigValid;
```

Comment:
If the config seems incomplete, then it seems like that should be a fatal errors. Other setup issues are treated as fatal errors (including using process.exit if necessary), and that seems to work well.